BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Authentication/ReflectionCredentialEnvelope.cs

```diff
@@ -33,6 +33,9 @@ public ReflectionCredentialEnvelope(object tokenCredential)
             if (IsValidType(tokenCredential))
             {
                 this.tokenRequestContext = AzureCore.MakeTokenRequestContext(scopes: AuthConstants.GetScopes());
+
+                // Call GetToken to initialize the internal cache.
+                _ = this.GetToken();
```

Comment:
This results in (probably) synchronous network I/O in the constructor. Are you sure you want to do that? Doesn't this get called as soon as you set the AzureTokenCredential on TelemetryConfiguration?

Reply:
Yes, this gets called when setting the Credential on TelmeteryConfiguration.
It was my intention to have this call during app startup, but I wasn't considering the worst-case scenario.

If you think it would be better to remove it I can do that.

Reply:
Unless we're relying on side-effects, yes I would definitely remove that because it will have startup performance impact.

Everything's lazy in the Azure SDKs I've used. You can construct a client with a TokenCredential cheaply, but no actual token fetching is done until you actually need it (via BearerTokenPipelinePolicy)